### ISSUE TITLE
Context always returns default value

### ISSUE THREAD

**Issue by adamaho** [solidjs/solid#882]
### Describe the bug

It seems that context is only ever returning the default value. 

I tweeted Ryan about this. Here is the thread.

https://twitter.com/theadamaho/status/1500483194117836802?s=20&t=R2WLCOtJHby_2IcfUTu8aA

Seems he could repro in the repo I sent but I am still having the issue. At this point I am kind of lost. Not sure if the issue is with my dev env at this point but I feel like this should just work.

### Your Example Website or App

https://github.com/adamaho/bonsai

### Steps to Reproduce the Bug or Issue

1. Follow the steps in the readme of the repo
2. Open the browser and look at the console to observe that context is undefined, even though it is clearly being created with a value.

### Expected behavior

As a user, I expect the console.log to show the value of the context that is provided, instead of undefined.

### Platform

- macOS (M1 air)
- Chrome

**Comment by ryansolid** [comment]
I see a problem with the current version. You are destructuring children in the components. This causes them to be evaluated outside of the context provider. Most notably here: https://github.com/adamaho/bonsai/blob/main/packages/libs/solid-button/src/button.tsx#L5

If you change all those components to use props with `props.children` inserted the Context should pass through.

**Comment by mdynnl** [comment]
There's also two different solid version , `1.3.11` and `1.3.8` in solid-button, not sure if these were the same before upgrade.

**Comment by adamaho** [comment]
From what I can see, each package.json has the same version of solid, `1.3.8` so that shouldn't be an issue.

re Ryan: I see. So I just changed back to `props.children`. I figured this was this issue but I wanted to confirm. Is this documented anywhere? I feel like this is going to slip up a lot of people that may be looking into solid and are coming from react. Destructuring function args is just a thing that I would expect to do without having to worry about breaking the functionality of the framework.

Thanks for your speedy response!
